.pylintrc

```diff
@@ -73,11 +72,41 @@ disable=spelling,       # way too noisy
     no-else-return,     # relax "elif" after a clause with a return
     docstring-first-line-empty, # relax docstring style
     import-outside-toplevel,
-    bad-continuation, bad-whitespace, # differences of opinion with black
-    import-error        # overzealous with our optionals/dynamic packages
-
-
-
+    import-error,        # overzealous with our optionals/dynamic packages
+
+    # TODO(#9614): these were added in modern Pylint. Decide if we want to enable them. If so,
+    #  remove from here and fix the issues. Else, move it above this section and add a comment
+    #  with the rationale.
+    arguments-renamed,
+    broad-exception-raised,
+    consider-iterating-dictionary,
+    consider-using-dict-items,
+    consider-using-enumerate,
+    consider-using-f-string,
+    consider-using-from-import,
+    modified-iterating-list,
+    nested-min-max,
+    no-member,
+    no-value-for-parameter,
```

Comment:
These two are generally important regular lints.  `no-member` has a habit of being a _little_ overzealous, but I'd be cautious about disabling it (even temporarily).  Ideally we'd catch all such errors with test coverage, but it's not 100%.